psi4/src/CMakeLists.txt

```diff
@@ -2,17 +2,13 @@
 ### >> This allow to propagate dependencies upwards sanely
 add_library(core SHARED core.cc)
 
-if(DEFINED ENV{LGTM_SRC})
-    target_include_directories(core
-      PRIVATE
-        ${Python_INCLUDE_DIRS}
-    )
-else()
-    target_link_libraries(core
-      PRIVATE
-        Python::Module
-    )
-endif()
+include_directories(
+    ${Python_INCLUDE_DIRS}
+)
+target_link_libraries(core
+  PRIVATE
+    Python::Module
+)
```

Comment:
did this come about b/c the build couldn't find Python.h at some point? using raw variables and general scope (e.g., `${Python_INCLUDE_DIRS}`) instead of targets (e.g., `Python::Module` and `core`) isn't the best practice. I've looked around https://pybind11.readthedocs.io/en/stable/compiling.html#advanced-interface-library-targets some and not fastened upon an ideal solution. `Python::Embed` is what we want to avoid iirc because that explicitly links core to libpython.so . a `target_include_directories(core SCOPE ${Python_INCLUDE_DIRS})` would be preferable if it could be here only or wherever the pybind11.h appeared only. This is rambling, so the main question is what was the error that this solved?

Reply:
Exactly.

```
In file included from /Users/jonathonmisiewicz/psi4/psi4/src/psi4/scfgrad/response.cc:50:
In file included from /Users/jonathonmisiewicz/psi4/psi4/src/psi4/libscf_solver/rhf.h:33:
In file included from /Users/jonathonmisiewicz/psi4/psi4/src/psi4/libscf_solver/hf.h:38:
In file included from /Users/jonathonmisiewicz/psi4/psi4/include/psi4/pybind11.h:38:
In file included from /Users/jonathonmisiewicz/miniconda3/envs/p4dev/include/pybind11/pybind11.h:44:
In file included from /Users/jonathonmisiewicz/miniconda3/envs/p4dev/include/pybind11/attr.h:13:
In file included from /Users/jonathonmisiewicz/miniconda3/envs/p4dev/include/pybind11/cast.h:13:
In file included from /Users/jonathonmisiewicz/miniconda3/envs/p4dev/include/pybind11/pytypes.h:12:
/Users/jonathonmisiewicz/miniconda3/envs/p4dev/include/pybind11/detail/common.h:128:10: fatal error: 'Python.h' file not found
#include <Python.h>
```

Reply:
You could try the below in libscf_solver/CMakeLists.txt. Otherwise, merge as it, and I can play with it later.

```
 target_include_directories(scf_solver_or_whatever
      PRIVATE
        ${Python_INCLUDE_DIRS}
    )
```

Reply:
I played with it myself and found that I needed to include it not only in `libdiis` and `libscf_solver` (both reasonable) but anything that imports `libscf_solver/hf.h`. That would be a nightmare for downstream. I'll leave this one for you to play with later.